maven-toolchain-builder/src/main/java/org/apache/maven/toolchain/discovery/JavaHomeFinderBasic.java

```diff
@@ -0,0 +1,410 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.maven.toolchain.discovery;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JavaHomeFinderBasic {
+    protected final Logger log = LoggerFactory.getLogger(getClass());
+    private final List<Supplier<? extends Set<String>>> myFinders = new ArrayList<>();
+
+    private boolean myCheckEmbeddedJava = false;
+
+    private String[] mySpecifiedPaths = new String[0];
+
+    public JavaHomeFinderBasic() {
+        myFinders.add(this::findInPATH);
+        myFinders.add(this::findInJavaHome);
+        myFinders.add(this::findInSpecifiedPaths);
+        myFinders.add(this::findJavaInstalledBySdkMan);
+        myFinders.add(this::findJavaInstalledByAsdfJava);
+        myFinders.add(this::findJavaInstalledByGradle);
+
```

Comment:
Linux and macOS have the Homebrew package manager. It allows installing various flavours of Java, too. Would it make sense to add support for that package manager?

Reply:
AFAIK, homebrew installs JDKs in the system directory, not in a homebrew specific directory, so that should  already be covered.

Reply:
Double checked and it does - at least, on macOS. I don't know for Linux.